ComponentKit/Core/Scope/CKThreadLocalComponentScope.mm

```diff
@@ -27,36 +27,46 @@ static pthread_key_t _threadKey() noexcept
   return thread_key;
 }
 
+static std::stack<CKThreadLocalComponentScope *> *_threadLocalStack() noexcept
+{
+  return static_cast<std::stack<CKThreadLocalComponentScope *> *>(pthread_getspecific(_threadKey()));
```

Comment:
This should work just fine, but another way to do this is to just grab the thread-local scope pointer and hold onto it in a member of CKThreadLocalComponentScope, then re-set it in the destructor.